cli/tests/test_concurrent_operations.rs

```diff
@@ -227,6 +227,101 @@ fn test_concurrent_snapshot_wc_reloadable() {
     ");
 }
 
+#[test]
+#[should_panic(expected = "Race condition detected")]
+fn test_git_head_race_condition() {
+    // Test for race condition where concurrent jj processes create divergent
+    // operations when importing/exporting Git HEAD. This test spawns two
+    // processes in parallel: one running `jj debug snapshot` repeatedly
+    // (which imports Git HEAD) and another running `jj next/prev` repeatedly
+    // (which exports Git HEAD). Without the fix, this would create divergent
+    // operations.
+
+    let test_env = TestEnvironment::default();
+    test_env
+        .run_jj_in(".", ["git", "init", "--colocate", "repo"])
+        .success();
+    let work_dir = test_env.work_dir("repo");
+
```

Comment:
nit: Can you insert a test snapshot that does actually "import git head" and use it in the assertion? This will prevent us from nullifying the test by changing the operation description.

```rust
// copy some bits from test_git_colocated.rs to do "import git head" ...
...
let output = work_dir.run_jj(["op", "log", "-T", "description", "--limit=1"]).success();
let desc = output.stdout.raw().lines().next().unwrap();
insta::assert_snapshot!(desc, @"import git head");
...
    .filter(|line| line.contains(desc))
```